### ISSUE TITLE
Feature request: Budget notifications

### ISSUE THREAD

**Issue by afeld** [terraform-providers/terraform-provider-aws#4548]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I'm using the new [`aws_budgets_budget`](https://www.terraform.io/docs/providers/aws/r/budgets_budget.html) resource, and am trying to set up notifications around it. This resource would support that use case.

I'm interested in implementing this, but would love feedback first.

### New or Affected Resource(s)

* aws_budgets_notification

### Potential Terraform Configuration

```hcl
data "aws_caller_identity" "current" {}

resource "aws_budgets_budget" "budget" {
  # ...
}

resource "aws_sns_topic" "team" {
  # ...
}

# OPTION 1

resource "aws_budgets_notification" "high_spending" {
  account_id = "${data.aws_caller_identity.current.account_id}"
  budget_name = "${aws_budgets_budget.budget.name}"

  notification_type = "ACTUAL"
  comparison_operator = "GREATER_THAN"
  threshold = "80"
  threshold_type = "PERCENTAGE"

  subscription_type = "SNS"
  subscription_address = "${aws_sns_topic.team.arn}"
}

# OPTION 2

resource "aws_budgets_notification" "high_spending" {
  account_id = "${data.aws_caller_identity.current.account_id}"
  budget_name = "${aws_budgets_budget.budget.name}"

  notification_type = "ACTUAL"
  comparison_operator = "GREATER_THAN"
  threshold = "80"
  threshold_type = "PERCENTAGE"

  subscription {
    type = "SNS"
    address = "${aws_sns_topic.team.arn}"
  }
}
```

#### Disclaimers

The subscription configuration needs to be included in the `aws_budgets_notification` resource configuration because [they are (strangely) required](https://docs.aws.amazon.com/aws-cost-management/latest/APIReference/API_budgets_CreateNotification.html#awscostmanagement-budgets_CreateNotification-request-Subscribers) when creating notifications, even though [there are API actions to add and remove them](https://docs.aws.amazon.com/aws-cost-management/latest/APIReference/API_Operations_AWS_Budgets.html).

Option 1 is presumably simpler to implement (in terms of parsing, state management, etc.), but Option 2 gives the option for supporting multiple `subscription` blocks.

The [account ID is required](https://docs.aws.amazon.com/aws-cost-management/latest/APIReference/API_budgets_CreateNotification.html#awscostmanagement-budgets_CreateNotification-request-AccountId) when creating a notification through the API, but it's not an option in the Console. It's unclear when that would/could be a value other than the "effective account ID".

### References

* #753

**Comment by flosell** [comment]
I opened a PR (#4523) a while back that's not merged yet, maybe add a 👍 over there if you are interested in this feature

**Comment by goochi1** [comment]
Was this ever added?

**Comment by bflad** [comment]
Support for `notification` configuration blocks in the `aws_budgets_budget` resource has been merged and will release with version 2.5.0 of the Terraform AWS Provider, later this week. Thanks, @flosell!

**Comment by Midacts** [comment]
Thanks for adding this functionality.
First time using this and noticed how recently this feature was added.
